crates/solver/src/driver.rs

```diff
@@ -200,11 +209,28 @@ impl Driver {
                 );
                 self.metrics
                     .settlement_submitted(err.as_outcome(), solver.name());
+                if let Some(transaction_hash) = err.transaction_hash() {
+                    if let Err(err) = self.metric_access_list_gas_saved(transaction_hash).await {
```

Comment:
Ah nice - so this will show us the "downside" of access lists if we revert early in a transaction and the access list causes us to pay for warm ups we don't actually use.

Reply:
Exactly. It should appear as a negative value of saved gas. Note that we also enter the function for cancelled transactions, but for those we don't use access list so calculation will be skipped.